### PR TITLE
Sagas

### PR DESCRIPTION
**Pull request by kalmanb**
First pass at Sagas

### CODE REVIEW

File: layout/rootSaga.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+import todos from '../todos'
+const { sagas } = todos
+
+const allSagas = [].concat(
+  sagas
+  // others ...
+)
+
+function * startSagas () {
```
Comment (d4ncer) [inline layout/rootSaga.js:12]: Why fork all sagas prior to injection?

File: todos/sagas.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+}
+
+// Fetch todos whenever receive a FETCH action
+function * watchTodos () {
+  console.log(actions.FETCH)
+
+  yield * takeEvery(actions.FETCH, getTodos)
+}
+
+const getRequest = () => new Promise((resolve, reject) => {
```
Comment (d4ncer) [inline todos/sagas.js:25]: Can we move this function into an api.js (name can be changed)?

File: layout/rootSaga.js (hunk 1 above)
Comment (kalmanb) [inline layout/rootSaga.js:12]: I think this is basically 'starting' them

File: todos/sagas.js (hunk 2 above)
Comment (kalmanb) [inline todos/sagas.js:25]: Yep - also want to move to superagent - get the backend up - get some nicely mocked tests etc.

**Pull request merged**